runtime/parachains/src/runtime_api_impl/v1.rs

```diff
@@ -231,6 +232,21 @@ pub fn session_index_for_child<T: initializer::Config>() -> SessionIndex {
 	<shared::Module<T>>::session_index()
 }
 
+pub fn get_historical_authority_ids<T: initializer::Config>() -> Vec<AuthorityDiscoveryId> {
+	let current_session_index = session_index_for_child::<T>();
+	let earliest_stored_session = <session_info::Module<T>>::earliest_stored_session();
```

Comment:
I imagine this might be racy and error-prone at session boundaries, we can maybe iterate directly over the `Sessions` storage, but that's probably racy as well

Reply:
@ordian  At session boundary, If there is a race condition between getting session indexes above and loop execution, we might miss latest session's authority ids, which can break the semantics. So, to cover widest possible range here we can query session info from earliest..=current+1.

Reply:
This should be fine at session boundaries as session changes are applied at the end of the parent block. The code comment in `fn session_index_for_child` is outdated and wrong and that threw me for a loop while investigating this comment.

Iterating up to `current_session_index` is fine but we also want to include the `AuthorityDiscvoery::next_authorities` here.